tests/integration/test_users.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Discourse integration tests."""
+
+import logging
+
+import pytest
+import requests
+from juju.action import Action
+from juju.application import Application
+from juju.unit import Unit
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+async def test_create_user(
+    app: Application,
+):
+    """
+    arrange: A discourse application
+    act: Create a user
+    assert: User is created, and re-creating the same user should fail
```

Comment:
This test is testing that when the action is run twice with a username, it fails. It is not really testing that a user is created. I believe a real assertion that the user is created could help.

Reply:
The second call of the action is verifying the user exists, through the rake call, which will only be successful if the user exists. I could include an API call to verify this, but i feel that this verification is covered in the rake task?

Reply:
The assertion in this case consists in running the function that is tested (again) and the test should name should be something like: "test_running_action_create_user_twice_fails". 

But this test is not testing the state of the system, I mean, that there is a user created in discourse.

Reply:
the test checks a user is created via the rake task

testing user creation is also verified in the following method